lib/livebook_web/live/session_live/app_teams_component.ex

```diff
@@ -169,29 +163,44 @@ defmodule LivebookWeb.SessionLive.AppTeamsComponent do
   end
 
   def handle_event("deploy_app", _, socket) do
-    notebook = Livebook.Session.get_notebook(socket.assigns.session.pid)
-
-    case Livebook.Teams.deploy_app(socket.assigns.hub, notebook, socket.assigns.session.files_dir) do
-      :ok ->
-        deployment_groups = Provider.deployment_groups(socket.assigns.hub)
-        deployment_group = Enum.find(deployment_groups, &(&1.id == notebook.deployment_group_id))
+    with {:ok, app_deployment} <- pack_app(socket),
+         :ok <- deploy_app(socket, app_deployment) do
+      message =
+        "App deployment for #{app_deployment.slug} with title #{app_deployment.title} created successfully"
 
-        app_deployment =
-          Enum.find(deployment_group.app_deployments, &(&1.slug == notebook.app_settings.slug))
+      {:noreply, assign(socket, messages: [{:info, message}])}
+    end
+  end
 
-        message =
-          "App deployment for #{app_deployment.slug} with title #{app_deployment.title} created successfully"
+  defp pack_app(socket) do
+    notebook = Livebook.Session.get_notebook(socket.assigns.session.pid)
+    files_dir = socket.assigns.session.files_dir
 
-        {:noreply, assign(socket, message_kind: :info, message: message, warnings: [])}
+    case Livebook.Teams.AppDeployment.new(notebook, files_dir) do
+      {:ok, app_deployment} ->
+        {:ok, app_deployment}
 
       {:warning, warnings} ->
-        {:noreply, assign(socket, warnings: warnings)}
+        warnings = Enum.map(warnings, &{:warning, &1})
+        {:noreply, assign(socket, messages: warnings)}
```

Comment:
One of the warnings that shouldn't allow the user to deploy is when the notebook stamp is invalid, as @jonatanklosko said in https://github.com/livebook-dev/livebook/pull/2511#discussion_r1525953027:

> There may be warnings, we should handle that. Currently the only warnings is if we cannot stamp the notebook correctly, so if there are any warnings we should rather show them to the user and not deploy.

Reply:
and building the App Deployment already when we show the modal is interesting, we would show the error if any of the attachments doesn't exist

Reply:
I see. So in my opinion it should really be an error, because the user cannot continue. So ignore my suggestion above, my new suggestion is: just remove `{:warning, _}` and return it as an error instead.